adding new conditioning for td modes

hectorestelles opened the pull request:
This PR adds a conditioning method for TD modes mimicking the standard behaviour of conditioning routines in LAL and GWSignal, i.e, starting the modes at a lower frequency for tapering not-in-band cycles and adding some extra buffer times.

stephengreen commented on dingo/gw/waveform_generator/waveform_generator.py:
```diff
@@ -1158,6 +1165,45 @@ def generate_TD_modes_L0(self, parameters):
+            {**parameters, "f_ref": self.f_ref}
+        )
+
+        f_min, new_fstart, textra, original_f_min, fisco = wfg_utils.get_starting_frequency_for_conditioning(parameters_gwsignal)
```
Keep to a standard variable name convention, e.g., `new_f_start`, `t_extra`, `f_isco`.

stephengreen commented on dingo/gw/waveform_generator/wfg_utils.py:
```diff
@@ -176,3 +176,80 @@ def get_polarizations_from_fd_modes_m(hlm_fd, iota, phase):
+def taper_td_modes_extra_time(h, extra_time, f_min, original_fmin, f_isco):
+
+    # Split in real and imaginary parts, since LAL conditioning routines are for real timeseries
+    h_tapered = lal.CreateREAL8TimeSeries(
```
Call this `h_tapered_re` for consistency with `h_tapered_im`.

stephengreen reviewed (changes requested):
Looks good, just a few minor comments on naming conventions and adding docstrings. Please run Black on all the files you changed.

I will commit an improvement to the test as well.

stephengreen commented:
I improved the mode reconstruction tests, including now a test for SEOBNRv5HM. When I ran the test on SEOBNRv5HM 1,000 times, I found mismatches up to 0.1. For SEOBNRv5PHM they stayed below 1e-7. Generally, mismatches are worse for HM than PHM.

stephengreen commented:
Also, since we now have a unit test for v5 waveforms, do we need to adjust the dependencies of Dingo, e.g., minimum version numbers to have v5 included?

hectorestelles commented:
Hi @stephengreen I tried to address the comments on commit https://github.com/dingo-gw/dingo/pull/217/commits/18c9b30ff77dd9da0bf170d95230dedc907893f2 

Let me know if you have further suggestions

stephengreen reviewed (commented):
Looks good. I updated the test tolerances and added a comment on the unit conversions.

Do we need to specify a minimum lalsimulation version for this test to pass?

hectorestelles commented:
> Looks good. I updated the test tolerances and added a comment on the unit conversions.
> 
> Do we need to specify a minimum lalsimulation version for this test to pass?

Yes, we would need to request lalsimulation>=7.15, in order to have gwsignal on it.
